File: java/NIM.java
import java.util.*;

public class NIM {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Masukkan NIM: ");
        String nim = input.nextLine();

        int[] digit = new int[nim.length()];

        // Pisahkan setiap karakter
        for (int i = 0; i < nim.length(); i++) {
            digit[i] = Character.getNumericValue(nim.charAt(i));
        }

        int total = 0;
        int maks = digit[0];
        int min = digit[0];

        for (int d : digit) {
            total += d;
            if (d > maks) maks = d;
            if (d < min) min = d;
        }

        double rata = (double) total / digit.length;

        // Reverse array
        System.out.print("\nReverse array: ");
        for (int i = digit.length - 1; i >= 0; i--) {
            System.out.print(digit[i] + " ");
        }

        System.out.println("\nTotal: " + total);
        System.out.println("Maksimum: " + maks);
        System.out.println("Minimum: " + min);
        System.out.println("Rata-rata: " + rata);
    }
}

File: php/nim.php
<?php

$nim = readline("Masukkan NIM: ");

$digit = [];

// Pisahkan setiap karakter ke dalam array
for ($i = 0; $i < strlen($nim); $i++) {
    $digit[] = (int)$nim[$i];
}

$total = array_sum($digit);
$max = max($digit);
$min = min($digit);
$rata = $total / count($digit);

// Reverse array
$reverse = array_reverse($digit);

echo "\nArray Reverse: ";
foreach ($reverse as $d) {
    echo $d . " ";
}

echo "\nTotal: $total";
echo "\nMaksimum: $max";
echo "\nMinimum: $min";
echo "\nRata-rata: $rata";

?>